iavl_orphaning_tree.go

```diff
@@ -73,10 +73,11 @@ func (tree *orphaningTree) SaveVersion(version uint64) {
 	// incorrectly marked as orphaned, since tree patterns after a re-balance
 	// may mirror previous tree patterns, with matching hashes.
 	tree.ndb.SaveBranch(tree.root, func(node *IAVLNode) {
-		tree.Unorphan(node.hash)
-
 		// The node version is set here since it isn't known until we save.
 		node.version = version
+		node.hashWithCount()
```

Comment:
but what?
we call hashWithCount in the callback, when it is called right after the callback in SaveBranch. twice?

Reply:
They're both necessary - this one is needed because Unorphan needs the updated hash. The one in SaveBranch is necessary because it can't rely on the caller to hash. In any case work is only done once, as hashWithCount is a noop if the node is already hashed.